packages/web/components/Dashboard/Settings/SubscriptionForm.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { toast } from 'react-toastify'
+import { loadStripe } from '@stripe/stripe-js'
+import { Elements, CardElement, useStripe, useElements } from '@stripe/react-stripe-js'
+import Button from '@/components/Button'
+import FormError from '@/components/FormError'
+import { useTranslation } from '@/config/i18n'
+import SettingsForm from '@/components/Dashboard/Settings/SettingsForm'
+import SettingsFieldset from '@/components/Dashboard/Settings/SettingsFieldset'
+import nProgress from 'nprogress'
+
+type FormValues = {
+  type: UserSubscriptionType
+  price: number
+}
+
+const stripeLib = loadStripe(process.env.STRIPE_PUBLIC_KEY!)
+
+const SubscriptionForm: React.FC = () => {
+  const { t } = useTranslation('settings.subscription')
+  const [customError, setCustomError] = useState()
+  const [customLoading, setCustomLoading] = useState(false)
+  const stripe = useStripe()
+  const elements = useElements()
+
+  const { handleSubmit, register, getValues, errors } = useForm<FormValues>({
+    mode: 'onSubmit',
+  })
+
+  const [createUserSubscription, { loading }] = useCreateUserSubscriptionMutation({
```

Comment:
Where `useCreateUserSubscriptionMutation` is coming?

Reply:
This will be an Apollo Hook generated based on the server-side `createMembershipSubscription` mutation (same re-naming as mentioned above).